### PR TITLE
BUG: DataFrame.all(bool_only=True) inconsistency with object dtype

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

Discussed on the dev call yesterday.

### CODE REVIEW

File: pandas/core/internals/blocks.py [hunk 1]
```diff
@@ -448,6 +448,20 @@ def f(mask, val, idx):
...
         return self.split_and_operate(None, f, inplace)
 
+    def _split(self) -> List["Block"]:
+        """
+        Split a block into a list of single-column blocks.
+        """
+        assert self.ndim == 2
+
+        new_blocks = []
+        for i, ref_loc in enumerate(self.mgr_locs):
```
Comment (jreback) [inline pandas/core/internals/blocks.py:458]: could be done as a list comprehension

Comment (jreback) [reply pandas/core/internals/blocks.py:458]: but nbd

**Review by jreback — APPROVED** [review]

File: doc/source/whatsnew/v1.2.0.rst [hunk 2]
```diff
@@ -235,6 +235,62 @@ Other enhancements
...
+.. _whatsnew_120.notable_bug_fixes:
+
+Notable bug fixes
+~~~~~~~~~~~~~~~~~
+
+These are bug fixes that might have notable behavior changes.
+
+Consistency of DataFrame Reductions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+:meth:`DataFrame.any` and :meth:`DataFrame.all` with ``bool_only=True`` now
```
Comment (jreback) [inline doc/source/whatsnew/v1.2.0.rst:249]: can you add this PR number here, and is there an actual issue? ping on green.

File: doc/source/whatsnew/v1.2.0.rst [hunk 3]
```diff
@@ -235,6 +235,62 @@ Other enhancements
...
+
+    In [6]: df[["B", "C"]].all(bool_only=True)
+    Out[6]:
+    B    False
+    C    True
+    dtype: bool
+
+*New behavior*:
+
+.. code-block:: ipython
```
Comment (jreback) [inline doc/source/whatsnew/v1.2.0.rst:279]: can you make this an ipython-block

File: pandas/core/internals/blocks.py [hunk 4]
```diff
@@ -450,6 +450,20 @@ def f(mask, val, idx):
 
         return self.split_and_operate(None, f, inplace)
 
+    def _split(self) -> List["Block"]:
+        """
+        Split a block into a list of single-column blocks.
+        """
+        assert self.ndim == 2
+
+        new_blocks = []
```
Comment (jreback) [inline pandas/core/internals/blocks.py:459]: might be worth adding some unit tests for this (e.g. on a block that is empty, since column and multilple)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/internals/blocks.py (hunk 4 above)
Comment (jbrockmendel) [reply pandas/core/internals/blocks.py:459]: good idea, was accidentally making a copy instead of a view!

**Comment by jbrockmendel** [comment]
ping

File: doc/source/whatsnew/v1.2.0.rst [hunk 5]
```diff
@@ -236,6 +236,54 @@ Other enhancements
...
+    Out[6]:
+    B    False
+    C    True
+    dtype: bool
+
+*New behavior*:
+
+.. ipython:: python
+
+    In [5]: df.all(bool_only=True)
```
Comment (jreback) [inline doc/source/whatsnew/v1.2.0.rst:282]: don't need the prompts (remove in followon / other PR)

File: pandas/core/internals/blocks.py (hunk 4 above)
Comment (jreback) [reply pandas/core/internals/blocks.py:459]: wow, testing is good!

**Review by jreback — APPROVED** [review]

File: doc/source/whatsnew/v1.2.0.rst (hunk 5 above)
Comment (jbrockmendel) [reply doc/source/whatsnew/v1.2.0.rst:282]: so remove the "In [5]: "?  do i also remove the "Out [5]: "?

Comment (jreback) [reply doc/source/whatsnew/v1.2.0.rst:282]: yep i python will generate appropriately

**Pull request merged**